### DISCUSSION TITLE
serving Public files in Vercel

### DISCUSSION THREAD

**Discussion by Wavesolid** [vercel/next.js#36214] [Help]
I'm using [Email.js](https://www.npmjs.com/package/emailjs) for sending Email function. But after I deployed the code to the Vercel, my code returns me an error 400 that unable to find  image in `public` path, As I tested before in my local, the `public` path was correct and Image was fetched properly. 
My Code (Error occured on line 73) :
![image](https://user-images.githubusercontent.com/58335667/163670716-e48ee978-8d15-4ef8-a3d7-629cf0a69e67.png)

The file photo is exist in my repositories aswell, but I'm lost what mistake that I did.
![image](https://user-images.githubusercontent.com/58335667/163670611-0f646cd5-63f7-4a0c-bea1-021b71a86c92.png)

**Comment by icyJoseph** [comment]
I think you don't need to use the `public` prefix, right?

According to the docs here: https://nextjs.org/docs/basic-features/static-file-serving:

```js
import Image from 'next/image'

function Avatar() {
  return <Image src="/me.png" alt="me" width="64" height="64" />
}

export default Avatar
```

Notice that it is `/me.png` and not `/public/me.png`.

Although, I wonder if here you should do something like `${process.cwd()}/public/example.png`. Not sure that's gonna cut it, but I'd start from there.

**Reply by Wavesolid** [reply]
@icyJoseph thanks for your respond. I already tried by not using `public` and directly access it to `/qrcodes/sadam.png` but the photo still not exist. Also I tried with `process.cwd()` on my path and give me a new error `/var/task/public/qrcodes/sadam.png does not exist`.
![image](https://user-images.githubusercontent.com/58335667/163705294-da51f80b-daf2-4cca-a6d1-63bd4ea1f9a7.png)

**Reply by icyJoseph** [reply]
Alright, I think I have understood the problem.

`Email.js` is actually checking for a file, and from within serverless functions that's not necessarily possible. It is, but not always. It depends on the serverless runtime.

From the `vercel` [docs I see that you need to specify which static assets should be made available to the runtime](https://vercel.com/support/articles/how-can-i-use-files-in-serverless-functions).

Furthermore, this is a [discussion here](https://github.com/vercel/next.js/discussions/32236) which I think you should go through.

It turns out this might not be so trivial. @thibautsabot perhaps you have more info on this than I do?

---

One thing I'd do, is host the images elsewhere, and fetch them into a stream, because `Email.js` supports streams as attachments.
